api/extension-guides/ai/ai-extensibility-overview.md

```diff
@@ -0,0 +1,146 @@
+
+MCP tools run outside of VS Code, either locally on the user's machine or as a remote service. Users can add MCP tools through JSON configuration or VS Code extension can configure them programmatically. You can implement MCP tools through various language SDKs and deployment options.
+
+As MCP tools run outside of VS Code, they do not have access to the VS Code extension APIs.
+
+**Key benefits**:
+
+- Domain-specific capabilities as part of an autonomous coding workflow
+- Automatic tool invocation in agent mode, depending on user intent
+- Local and remote deployment options
+- Share implementations across different tools and platforms
+
+**Key considerations**:
+
+- No access to VS Code extension APIs
+- Distribution and deployment require users to set up the MCP server
+
+### Chat participant
+
+Chat participants are specialized assistants that enable users to extend ask mode with domain-specific experts. In ask mode, users can invoke a chat participant by @-mentioning it and passing in a natural language prompt about a particular topic or domain. The chat participant is responsible for handling the entire chat interaction.
+
+To implement a chat participant, use the [Chat API](/api/extension-guides/ai/chat) within your VS Code extension. A chat participant can access all VS Code extension APIs and provide deep integration with the editor.
+
+**Key benefits**:
+
+- Provide domain-specific expertise in ask mode
+- Control interaction flow and customize response behavior
+- Deep VS Code integration through extension APIs
+- Easy distribution and deployment via the Visual Studio Marketplace
+
+**Key considerations**:
+
+- Only available in ask mode
+- Remote deployment requires the extension to implement the client-server communication
+- Reuse across different tools requires modular design and implementation
+
+## Build your own AI-powered features
+
+VS Code gives you direct programmatic access to AI models for creating custom AI-powered features in your extensions. This approach enables you to build editor-specific interactions that use AI capabilities without relying on the chat interface.
+
+To use language models directly, use the [Language Model API](/api/extension-guides/ai/language-model) within your VS Code extension. You can incorporate these AI capabilities into any extension feature, such as code actions, hover providers, custom views, and more.
+
+**Key benefits**:
+
+- Create custom workflows and experiences that target specific use cases
+- Integrate AI capabilities into existing extension features
+- Deep VS Code integration through extension APIs
+- Easy distribution and deployment via the Visual Studio Marketplace
+
+**Key considerations**:
+
+- Reuse across chat or different tools requires modular design and implementation
+
+## Decide which option to use
+
+When choosing the right approach for extending AI in your VS Code extension, consider the following guidelines:
+
+1. **Choose Language Model Tool when**:
```

Comment:
I think this list is another reason why we can cut down on the Key Benefits in the previous section. This again feels like repetition.

Reply:
I've trimmed and consolidated the list of benefits for each section.

Reply:
Thank you. Marking as resolved.